src/client/api/metrics.c

```diff
@@ -0,0 +1,914 @@
+{
+	int rc = 0;
+	D_ALLOC_PTR(*stats);
+	if (*stats == NULL) {
+		D_ERROR("Failed to allocate memory\n");
+		rc = -DER_NOMEM;
+	}
+	return rc;
+}
+
+int
+daos_metrics_free_statsbuf(daos_metrics_ustats_t *stats)
+{
+	free(stats);
+	return 0;
+}
+
+int
+daos_metrics_get_stats(enum daos_metrics_stats_grp ms_grp, daos_metrics_ustats_t *stats)
+{
+	int rc = 0;
+	dc_metrics_tls_data_t *entry;
+	if (is_metrics_enabled == 0) {
+		D_GOTO(out, rc = 0);
+	}
+	if ((ms_grp != DAOS_METRICS_OBJ_UPDATE_STATS) && \
+	    (ms_grp != DAOS_METRICS_OBJ_FETCH_STATS)) {
+		D_ERROR("Invalid argument ms_grp = %d\n", ms_grp);
+		D_GOTO(out, rc = -DER_INVAL);
+	}
+
+	D_MUTEX_LOCK(&metrics_tls_lock);
+	memset(stats, 0, sizeof(*stats));
+	if (ms_grp == DAOS_METRICS_OBJ_UPDATE_STATS) {
+		stats->ms_grp = DAOS_METRICS_OBJ_UPDATE_STATS;
+		stats->u.st_obj_fetch.st_min = ULONG_MAX;
+	} else if (ms_grp == DAOS_METRICS_OBJ_FETCH_STATS) {
+		stats->ms_grp = DAOS_METRICS_OBJ_FETCH_STATS;
+		stats->u.st_obj_update.st_min = ULONG_MAX;
+	}
+	d_list_for_each_entry(entry, &metrics_list, list) {
+		if (ms_grp == DAOS_METRICS_OBJ_UPDATE_STATS) {
+			stats->u.st_obj_update.st_value += entry->update_stat.st_value;
+			if (stats->u.st_obj_update.st_min > entry->update_stat.st_min)
+				stats->u.st_obj_update.st_min = entry->update_stat.st_min;
+			if (stats->u.st_obj_update.st_max < entry->update_stat.st_max)
+				stats->u.st_obj_update.st_max = entry->update_stat.st_max;
+			stats->u.st_obj_update.st_sum += entry->update_stat.st_sum;
+			stats->u.st_obj_update.st_sum_of_squares += \
+				entry->update_stat.st_sum_of_squares;
+		} else if (ms_grp == DAOS_METRICS_OBJ_FETCH_STATS) {
+			stats->u.st_obj_fetch.st_value += entry->fetch_stat.st_value;
+			if (stats->u.st_obj_fetch.st_min > entry->fetch_stat.st_min)
+				stats->u.st_obj_fetch.st_min = entry->fetch_stat.st_min;
+			if (stats->u.st_obj_fetch.st_max < entry->fetch_stat.st_max)
+				stats->u.st_obj_fetch.st_max = entry->fetch_stat.st_max;
+			stats->u.st_obj_fetch.st_sum += entry->fetch_stat.st_sum;
+			stats->u.st_obj_fetch.st_sum_of_squares += \
+				entry->fetch_stat.st_sum_of_squares;
+		}
+	}
+	D_MUTEX_UNLOCK(&metrics_tls_lock);
+	if (stats->u.st_obj_update.st_min == ULONG_MAX) {
+		stats->u.st_obj_update.st_min = 0;
+	}
+	if (stats->u.st_obj_fetch.st_min == ULONG_MAX) {
+		stats->u.st_obj_fetch.st_min = 0;
+	}
+out:
+	return rc;
+}
+
+int
+daos_metrics_alloc_distbuf(daos_metrics_udists_t **dists)
+{
+	int rc = 0;
+	D_ALLOC_PTR(*dists);
+	if (*dists == NULL) {
+		D_ERROR("Failed to allocate memory\n");
+		rc = -DER_NOMEM;
+	}
+	return rc;
+}
+
+int
+daos_metrics_free_distbuf(daos_metrics_udists_t *dists)
+{
+	free(dists);
+	return 0;
+}
+
+int
+daos_metrics_get_dist(enum daos_metrics_dist_grp md_grp, daos_metrics_udists_t *dist)
+{
+	int rc = 0, i;
+	dc_metrics_tls_data_t *entry;
+
+	if (is_metrics_enabled == 0) {
+		D_GOTO(out, rc = 0);
+	}
+	memset(dist, 0, sizeof(*dist));
+	D_MUTEX_LOCK(&metrics_tls_lock);
+	switch(md_grp) {
```

Comment:
(style)  switch and case should be at the same indent
(style)  space required before the open parenthesis '('